### DISCUSSION TITLE
Interested in working on " Attachment support for Kart" project from GSOC ideas list.

### DISCUSSION THREAD

**Discussion by Zumitify** [koordinates/kart#607] [Google Soc / SoD]
Hi folks of the community,
My name is [Sumit](https://www.linkedin.com/in/sumit-p/). I am currently working as a SDE and have completed my bachelor's in Computer Engineering in May 2020. 
I have managed to [build](https://github.com/koordinates/kart/blob/master/CONTRIBUTING.md#building-the-development-version-with-cmake-windows) Kart locally and I am currently exploring the product, codebase & running test cases.

### Problem statement: Attachment support for Kart
I have gone over the issue [details](https://github.com/koordinates/kart/issues/583) and the high-level objective of the project is clear to me. My technical skills are aligned with the requirements and I am well versed in working with large codebases.

> [Big job](https://github.com/koordinates/kart/issues/583#issue-1170483738). The following existing Git technologies may be useful here: index files, and sparse checkouts
> A good [first step](https://github.com/koordinates/kart/issues/583#issuecomment-1068690985) would be to simply detect and log which attachments have changed between two commits

 As my next step, I plan to explore the internal working of `kart diff` and read about `index files and sparse checkouts etc` .

-  @olsen232 Would you suggest any other topics/resources that I should go over which will help me with this project?

Having been a [cartophile](https://en.wiktionary.org/wiki/cartophile), GSOC + Kart can provide me a good opportunity to kickstart my open source journey. I look forward to hearing from you and working on this project.
Thanks.

**Comment by rcoup** [comment]
Hi Sumit,

Sorry about the delay in getting back to you. The [Pro Git book](https://git-scm.com/book/en/v2) and the [Git documentation](https://git-scm.com/docs) probably have the best general overview of the concepts, and for even more technical specifics there are deep-dive blog posts you will find on the web. Kart _generally_ works the same as Git, except for our particular focus on "datasets" and "features/rows" which is different from how Git works.

**Reply by Zumitify** [reply]
Hi Robert, thanks for the reply. I will surely go through the resources suggested.

**Reply by rcoup** [reply]
Ah, I saw your Google doc and added some comments there.
